tests/slostatusapi_test.go

```diff
@@ -0,0 +1,275 @@
+//go:build e2e_test
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v1alphaExamples "github.com/nobl9/nobl9-go/internal/manifest/v1alpha/examples"
+	"github.com/nobl9/nobl9-go/manifest"
+	"github.com/nobl9/nobl9-go/manifest/v1alpha"
+	v1alphaDirect "github.com/nobl9/nobl9-go/manifest/v1alpha/direct"
+	v1alphaService "github.com/nobl9/nobl9-go/manifest/v1alpha/service"
+	v1alphaSLO "github.com/nobl9/nobl9-go/manifest/v1alpha/slo"
+	v1 "github.com/nobl9/nobl9-go/sdk/endpoints/slostatusapi/v1"
+	v2 "github.com/nobl9/nobl9-go/sdk/endpoints/slostatusapi/v2"
+)
+
+func Test_SLOStatusAPI_V1_GetSLO(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	allObjects := setupSLOListTest(t)
+	project, _, slo := allObjects[0], allObjects[1], allObjects[2]
+
+	v1Apply(t, allObjects)
+	t.Cleanup(func() { v1Delete(t, allObjects) })
+
+	var err error
+	var response v1.SLODetails
+	ticker := time.NewTicker(5 * time.Second)
+	timer := time.NewTimer(time.Minute)
+	defer ticker.Stop()
+	defer timer.Stop()
+
+	done := false
+	for !done {
+		select {
+		case <-ticker.C:
+			response, err = client.SLOStatusAPI().V1().GetSLO(ctx, slo.GetName(), project.GetName())
+			if err == nil {
+				done = true
+			}
+		case <-timer.C:
+			t.Error("timeout")
+		}
+	}
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, response)
+	assert.Equal(t, slo.GetName(), response.Name)
+}
```

Comment:
you can limit the boilerplate of executing these requests until timeout or success like that:

```suggestion
func Test_SLOStatusAPI_V1_GetSLO(t *testing.T) {
	t.Parallel()
	ctx := context.Background()

	allObjects := setupSLOListTest(t)
	project, _, slo := allObjects[0], allObjects[1], allObjects[2]

	v1Apply(t, allObjects)
	t.Cleanup(func() { v1Delete(t, allObjects) })

	response, err := tryExecuteSLOStatusAPIRequest(t, func() (v1.SLODetails, error) {
		return client.SLOStatusAPI().V1().GetSLO(ctx, slo.GetName(), project.GetName())
	})

	require.NoError(t, err)
	assert.NotEmpty(t, response)
	assert.Equal(t, slo.GetName(), response.Name)
}

func tryExecuteSLOStatusAPIRequest[T any](t *testing.T, reqFunc func() (T, error)) (T, error) {
	t.Helper()
	ticker := time.NewTicker(5 * time.Second)
	timer := time.NewTimer(time.Minute)
	defer ticker.Stop()
	defer timer.Stop()

	var (
		response T
		err      error
	)
	for {
		select {
		case <-ticker.C:
			response, err = reqFunc()
			if err == nil {
				return response, nil
			}
		case <-timer.C:
			t.Error("timeout")
			return response, err
		}
	}
}
```